rts/Game/Camera/SpringController.cpp

```diff
@@ -75,9 +115,14 @@ void CSpringController::KeyMove(float3 move)
 		return;
 	}
 
+	const float3 prevPos = pos;
+
 	move *= 200.0f;
 	const float3 flatForward = (dir * XZVector).ANormalize();
 	pos += (camera->GetRight() * move.x + flatForward * move.y) * pixelSize * 2.0f * scrollSpeed;
+
+	SmoothCamHeight(prevPos);
```

Comment:
How about, instead of the mode check inside `SmoothCamHeight()` in line 88, something like this?
```cpp
	switch (trackMapHeight) {

	case HeightTracking::Disabled:
		/* also calls SmoothCamHeight, because
		 * of reason <FIXME: fill in reason> */
		[[fallthrough]];

	case HeightTracking::Smooth:
		SmoothCamHeight(prevPos);
		break;

	case HeightTracking::Terrain:
		/* tracking the terrain is handled at <FIXME: where?>,
		 * so we don't do anything here */
		break;
	}
```

Reply:
that's much better